sdk/core/azure-core/inc/azure/core/operation.hpp

```diff
@@ -51,6 +51,8 @@ namespace Azure { namespace Core {
      */
     virtual std::string GetResumeToken() const = 0;
 
+    virtual Azure::Core::Http::RawResponse* GetRawResponse() const = 0;
```

Comment:
Need doc comments.

Reply:
Should this be as follows to match `Response<T>`?
https://github.com/Azure/azure-sdk-for-cpp/blob/3b88c7bb83b04382f5eb1fbf62a1bbecd6ec3620/sdk/core/azure-core/inc/azure/core/response.hpp#L48-L53

```suggestion
    virtual Azure::Core::Http::RawResponse& GetRawResponse() const = 0;
```

Reply:
Add  a changelog entry.

Reply:
Will do.  Though customers should NEVER use the type.  It should be moved to internal as well.

Reply:
`Operation<T>`? Really? That's surprising then that .NET put it at the root namespace:
https://github.com/Azure/azure-sdk-for-net/blob/3e8a709af4c78c856cc753a8ed5a48c95a1fa92a/sdk/core/Azure.Core/src/OperationOfT.cs#L11-L21

Reply:
Yea, C++ has value semantics and that means that if a user can create an operation<T> they could then replace the guts of the derived type.  Remember that in .NET<T> its a generic type which is not the same as a template.

Reply:
Are we going to return a pointer or a reference? `Response<T>` returns a reference.

Reply:
+1 for reference and maybe we should have two overloads

```
virtual Azure::Core::Http::RawResponse const& GetRawResponse() const = 0;
virtual Azure::Core::Http::RawResponse& GetRawResponse() = 0;
```

Reply:
If we do a reference return then we need to throw if the ptr is null.  Otherwise its undefined behavior.  

What do we do if a user extracts the m_rawResponse and then calls GetRawResponse?  Users need to try/catch all calls to GetRawResponse because they may or may not have access to the rawresponse.

Reply:
> If we do a reference return then we need to throw if the ptr is null. Otherwise its undefined behavior.
> 
> What do we do if a user extracts the m_rawResponse and then calls GetRawResponse? Users need to try/catch all calls to GetRawResponse because they may or may not have access to the rawresponse.

same question for `Response<T>`

Reply:
Well, we solved this on Nullable<T> by aborting if T is empty and we call Get().
Probably we should do the same with Response<T> and Operation<T>

Reply:
At least this should return a pointer to a constant RawResponse?  or do we want to allow to update the RawResponse?